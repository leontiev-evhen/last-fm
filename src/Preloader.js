import React from 'react';

export default function Preloader() {
  return (
    <div className="preloder">
      <div className="loader"></div>
    </div>
  );
}