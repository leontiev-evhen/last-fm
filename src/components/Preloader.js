import React from 'react';

export function Preloader() {
  return (
    <div className="preloder">
      <div className="loader"></div>
    </div>
  );
}