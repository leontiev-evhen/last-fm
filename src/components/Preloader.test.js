import React from 'react';
import { mount } from 'enzyme';
import { Preloader } from './Preloader';

describe('Preloader component', () => {
  const component = mount(<Preloader />);
  it('render correctly Preloader component', () => {
    expect(component).toMatchSnapshot();
  });
});