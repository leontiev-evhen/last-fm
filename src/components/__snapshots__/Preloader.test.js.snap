// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader component render correctly Preloader component 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Preloader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="preloder"
      >
        <div
          class="loader"
        />
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="loader"
        />,
        "className": "preloder",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="loader"
          />,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "loader",
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="preloder"
        >
          <div
            class="loader"
          />
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="loader"
          />,
          "className": "preloder",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="loader"
            />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "loader",
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
